### PR TITLE
[Monitor] Add Public Preview APIs of Metric Baseline

### PR DESCRIPTION
**Pull request by asafst**
These APIs include:
1. Getting the pre-calculated baseline of a metric
2. Calculating the baseline of a metric given the metric values

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by sergey-shandar** [comment]
@simongdavies could you have a look?

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+        "flow": "implicit",
+        "description": "Azure Active Directory OAuth2 Flow",
+        "scopes": {
+          "user_impersonation": "impersonate your user account"
+        }
+      }
+    },
+    "paths": {
+      "/{resourceUri}/providers/microsoft.insights/baseline/{metricName}": {
```
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: Description should state/show examples of resourceUri

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+          "message": {
+            "description": "Error message indicating why the operation failed.",
+            "type": "string"
+          }
+        }
+      },
+      "BaselineResponse": {
+        "type": "object",
+        "properties": {
+          "id": {
```
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: id, name and type properties should be marked as readonly: true

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+          "name": {
+            "$ref": "#/definitions/LocalizableString",
+            "description": "the name and the display name of the metric, i.e. it is localizable string."
+          },
+          "properties": {
+            "x-ms-client-flatten": true,
+            "$ref": "#/definitions/BaselineProperties",
+            "description": "the properties of the baseline."
+          },
+          "timestamps": {
```
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: timestamps, baseline, metadata should be under properties envelope.

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+            "description": "the baseline metadata values."
+          }
+        }, 
+        "required": [ "id", "type", "name" ],
+        "description": "The response to a baseline query."
+      },
+      "Baseline": {
+        "type": "object",
+        "properties": {
+          "sensitivity": {
```
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: enum?

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+          "localizedValue": {
+            "type": "string",
+            "description": "the locale specific value."
+          }
+        },
+        "description": "The localizable string class."
+      },
+      "BaselineProperties": {
+        "properties": {
+          "timespan": {
```
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: date-time format?

Comment (asafst) [reply specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: It should be an ISO 8601 time interval. The format is "{date-time}/{date-time}". So string is probably more appropriate here.
I tried to be consistent with the definitions and types in the metrics_API.

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json (hunk 1 above)
Comment (asafst) [reply specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: Added the description to the ResourceUriParameter parameter part.

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json (hunk 2 above)
Comment (asafst) [reply specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: Added readonly:true

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json (hunk 3 above)
Comment (asafst) [reply specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: Moved them to properties

File: specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json (hunk 4 above)
Comment (asafst) [reply specification/monitor/resource-manager/microsoft.insights/2017-11-01-preview/baseline_API.json]: You're right, changed to enum

**Comment by ravbhatnagar** [comment]
Looks good.

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**